### ISSUE TITLE
MySQL strict mode compatibility

### ISSUE THREAD

**Issue by zonky2** [MetaModels/attribute_alias#28]
with MySQL strict mode we have the follow error

```
An exception occurred while executing 'UPDATE mm_employees SET alias = ? WHERE id = ?' with params [null, 80]:

SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'alias' cannot be null
```
we can include `serializeData` with

```
    public function serializeData($value)
    {
        return $value === null ? '' : $value;
    }
```

or we change `getSQLDataType` and set default to `NULL`

btw: if we change the default from '' to NULL, we must write a updating script - or not?

**Comment by richardhj** [comment]
Allowing `null` should be fine.

**Comment by zonky2** [comment]
> Allowing

hmm? 
is not the question whether we set NULL or '' as default? and not whether we allow what or not...

**Comment by zonky2** [comment]
discussed with @discordier: we set default to `NULL` - also for other text based attributes; we have to create an updater yet

**Comment by Ainschy** [comment]
there is a issue when you use force_alias

`UPDATE mm_metamodel SET alias = ? WHERE id = ?' with params [null, 5]: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'alias' cannot be null`

**Comment by discordier** [comment]
@Ainschy This is already fixed in #29, we need the migration though to finish it. This is on hold until we have https://github.com/contao-community-alliance/contao-polyfill-bundle/pull/2 out of the door.

**Comment by zonky2** [comment]
fixed
